### ISSUE TITLE
Add metadata from datapackage

### ISSUE THREAD

**Issue by khelfen** [htw-pv3/weather-data#4]
#### Empty Files v1.4 (to be filled)

 - [x] db_log @khelfen 
 - [x] htw_wetter_sun_2015
 - [x] htw_wetter_time_2015

#### Information v1.3 (filled with information but old version)

***Do not do this anymore***
* ***Clone my Branch and create your own!***
* You can find the v1.3 inside [htw_pv3_postgresql_05_setup_tables_and_import_data.sql](https://github.com/htw-pv3/weather-data/blob/master/postgresql/htw_pv3_postgresql_05_setup_tables_and_import_data.sql).
* Please update the linked .json data into the [v1.4 format](https://github.com/OpenEnergyPlatform/metadata/tree/develop/metadata/v140).

### Finish with a Pull Request on my [Branch](https://github.com/htw-pv3/weather-data/tree/MyMetaData/helfenbein_kilian) and me as reviewer:

 - [x] db_log
 - [x] htw_wetter_sun_2015
 - [x] htw_wetter_time_2015

- [x] transfer into [htw_pv3_postgresql_05_setup_tables_and_import_data.sql](https://github.com/htw-pv3/weather-data/blob/master/postgresql/htw_pv3_postgresql_05_setup_tables_and_import_data.sql) @khelfen 

### Miscellaneous

Please ***assign yourself per edit directly into this post*** with your initials like @khelfen regarding the files before you start working on them to prevent redundance!

**Comment by nailend** [comment]
@khelfen the json file on your branch already seem to be v1.4 but empty/half emtpy. 
At least it says "metadataVersion": "OEP-1.4.0"

**Comment by khelfen** [comment]
Sorry for making it unclear! Yes, the files are v1.4 but are empty! The v1.3 is inside the [htw_pv3_postgresql_05_setup_tables_and_import_data.sql](https://github.com/htw-pv3/weather-data/blob/master/postgresql/htw_pv3_postgresql_05_setup_tables_and_import_data.sql)

**Comment by khelfen** [comment]
Oh! I see my mistake now. Yes they are filled correctly but not inside the sql! Will fix it tomorrow!

**Comment by Ludee** [comment]
You just found an ongoing problem with metadata in the _.json_ file and the corresponding sql command (COMMENT ON TABLE) in the scripts.
There are two version which have to be maintained. I only updated the _datapackage_.
A perfect solution would be to read the metadata from file, but this doesn't work. (In Python this would be no problem btw.)

**Comment by Ludee** [comment]
The branch is way too big! They all deal with metadata, but are in different files.

**Comment by khelfen** [comment]
@Ludee Yes I already wanted to ask if there is a way to solve this with directly reading from the .json. Too bad.

I opened a new branch as suggested in #11. What should I do with my old Branch? Not sure what to do atm.

**Comment by Ludee** [comment]
Thanks for the quick response! The goal is to improve and learn and you are fast.
I prefer closing and making a new branch and avoid to rename a branch.

Now there is an interesting situation between the branches:

Master contains the "provided metadata" in the folder "metadata".
Your branch is behind and only contains the "contributed metadata" in a folder "metadata-A4".
Adding specific information to names (files, branches...) is not so good, since this info is already in the git history. In a merge, we would have two folders with metadata...

So my suggestion is to rename the folder to "metadata" as well on your branch. Then you can pull the existing information from master-branch in your branch and your "feature" is up-to-date and easy to merge in a PR.

**Comment by khelfen** [comment]
@Ludee having trouble with this

* renamed the folder
* trying to `git pull origin master` and getting an error as expected:

![grafik](https://user-images.githubusercontent.com/46815982/81318130-0ca33600-908e-11ea-9590-122339703ad7.png)

* at this point it just deletes the file einleuchtend_wrdata_2015_wr5.json without giving me a choice
* used the mergetool meld afterwards for the other files (which I failed at first but made it work after a while) but wr5 is just missing
* I don't know why atm

**Comment by khelfen** [comment]
@Ludee Made a workaround by manually copy/pasting the files (I know, I know...) as I didn't find any reason for the behaviour described above. Unfortunately my PR has merge conflicts now. 

Would be great if we could discuss what would have been the right way in our next lesson!

**Comment by nailend** [comment]
> You just found an ongoing problem with metadata in the _.json_ file and the corresponding sql command (COMMENT ON TABLE) in the scripts.
> There are two version which have to be maintained. I only updated the _datapackage_.
> A perfect solution would be to read the metadata from file, but this doesn't work. (In Python this would be no problem btw.)

could you please be more specific and link a line please?

**Comment by Ludee** [comment]
> could you please be more specific and link a line please?

To be more specific, there are two versions of the metadata in this repo now. I use the weather data as an example.

1. [A JSON file containing only the metadata](https://github.com/htw-pv3/weather-data/blob/master/metadata/htw_wetter_weatherdata_2015.json)

1.  [An SQL function _COMMENT ON TABLE_ that writes the metadata to the database](https://github.com/htw-pv3/weather-data/blob/master/postgresql/htw_pv3_postgresql_05_setup_tables_and_import_data.sql#L32-L237)

If we update the metadata (add infos, fix a typo) we need to do it twice!

**Comment by mauricio-ocampo** [comment]
@Ludee SO you mean that the metadata from [1.](https://github.com/htw-pv3/weather-data/blob/master/metadata/htw_wetter_weatherdata_2015.json) we should call it with a function in [2.](https://github.com/htw-pv3/weather-data/blob/master/postgresql/htw_pv3_postgresql_05_setup_tables_and_import_data.sql#L32-L237) in order to avoid dople changes, that means we just have to update the 1. json file and automatically will be updated in 2. if that´s right, do you think this [webpage](https://www.pauric.blog/How-To-Import-JSON-To-SQL-Server/) helps? with the function OPENJSON?

**Comment by mauricio-ocampo** [comment]
@Ludee @khelfen @nailend the [Skript 4](https://github.com/htw-pv3/weather-data/blob/master/postgresql/htw_pv3_postgresql_04_function_db_log.sql) has in the lines 254-256 the "model" as PV3 and "version_name" also as PV3. Is it that OK? also in the Line 579 PV3 is as "Project" designated? does SQL differentiate Capital letters (uppercase) and small letters (lowercase), ´cause there are several PV3 and pv3. PV3 was before MA3

**Comment by nailend** [comment]
@mauricio-ocampo Yes I changed it to PV3 as this is the name of our project but I did not modify skript 6 yet.  I assume, SQL will differentiates capital letters depending on the input type. In case of a string ("PV3","pv3") they will probably make a difference, especially if they are written as such into a table. Concerning the commands it is of readabiltys sake [see here] (https://stackoverflow.com/questions/9325927/why-are-mysql-queries-almost-always-written-in-capital)
